Godeps/_workspace/src/github.com/evanphx/json-patch/patch.go

```diff
@@ -314,7 +325,38 @@ func (d *partialArray) set(key string, val *lazyNode) error {
 	}
 
 	idx, err := strconv.Atoi(key)
+	if err != nil {
+		return err
+	}
+
+	sz := len(*d)
+	if idx+1 > sz {
+		sz = idx + 1
+	}
+
+	ary := make([]*lazyNode, sz)
+
+	cur := *d
+
+	copy(ary, cur)
+
+	if idx >= len(ary) {
+		fmt.Printf("huh?: %#v[%d] %s, %s\n", ary, idx)
```

Comment:
This version has been merged to Kubernetes now with the beautiful `huh` error message. Can we merge this as well? It's a defensive check that should never really be hit. I've submitted a PR to json-patch but not merged yet & would really like to get this merged.